pm_yaml_config/pm_yaml_config.xml

```diff
@@ -0,0 +1,126 @@
+                    <option value="3ob-3-1" selected="true">3ob-3-1</option>
+                    <option value="pbc-0-3">pbc-0-3</option>
+                </param>
+                <param type="text" name="dftb_optionals" value="[]" optional="true" label="dftb optionals" help="additional optional json files to activate for DFTBArgs (for example, dftd3.json will use DFTD3 dispersion forces for 3ob-3-1 if DFTB+ has been compiled to support them)."/>
+                <param type="select" name="dftb_pbc" optional="true" value="true" label="dftp pbc" help="whether to turn on periodic boundary conditions in DFTB+.">
+                    <option value="true" selected="true">true</option>
+                    <option value="false">false</option>
+                </param>
+            </when>
+            <when value="all">
+                <param type="float" name="uep_gw_factor" value="4.0" optional="true" label="gaussian witdh factor" help="Gaussian width factor for UEP calculation. Higher values will make the potential of atomic nuclei closer to the point-like approximation but may introduce artifacts."/>
+                <param type="text" name="uep_chden" optional="true" label="path to .den_fmt and castep files" help="path to the .den_fmt file containing the electronic density for an Unperturbed Electrostatic Potential optimisation. The corresponding .castep file must be in the same folder and with the same seedname."/>
+                <param type="text" name="castep_command" optional="true" label="castep command" help="command to use to run CASTEP."/>
+                <param type="text" name="castep_param" optional="true" label="PARAM path" help="file path to the CASTEP parameter file."/>
+                <param type="text" name="mu_symbol" optional="true" label="mu symbol" help="the symbol to use for the muon when writing out the castep custom species." />
+                <param type="text" name="dftb_set" optional="true" label="dftb set" help="the parameter set to use for DFTB+. Currently supported are 3ob-3-1 and pbc-0-3. For more information see (the DFTB site)[http://www.dftb.org/parameters/]."/>
+                <param type="text" name="dftb_optionals" value="[]" optional="true" label="dftb optionals" help="additional optional json files to activate for DFTBArgs (for example, dftd3.json will use DFTD3 dispersion forces for 3ob-3-1 if DFTB+ has been compiled to support them)."/>
+                <param type="select" name="dftb_pbc" value="true" optional="true" label="dftp pbc" help="whether to turn on periodic boundary conditions in DFTB+.">
+                    <option value="true">true</option>
+                    <option value="false">false</option>
+                </param>
+            </when>
+        </conditional>
+        <param type="text" name="clustering_method" label="clustering method" value="hier" />
```

Comment:
This should be a `select` with options `hierarchical` and `k-means` (corresp. `hier` and `kmeans` in the YAML).
The `clustering_hier_t` (float) and `clustering_kmeans_k` (int) params should be conditionally required based on the option selected here.

(the k in kmeans has nothing to do with the k in k-point grid 😬 )

Reply:
you mean k-points grid?

Reply:
changed